pkg/virt-config/config-map.go

```diff
@@ -228,17 +228,33 @@ func (c *ClusterConfig) SetConfigModifiedCallback(cb ConfigModifiedFn) {
 	go c.configModifiedCallback()
 }
 
+// This struct is for backward compatibility and is deprecated, no new fields should be added
+type migrationConfiguration struct {
+	NodeDrainTaintKey                 *string            `json:"nodeDrainTaintKey,omitempty"`
+	ParallelOutboundMigrationsPerNode *uint32            `json:"parallelOutboundMigrationsPerNode,string,omitempty"`
+	ParallelMigrationsPerCluster      *uint32            `json:"parallelMigrationsPerCluster,string,omitempty"`
+	AllowAutoConverge                 *bool              `json:"allowAutoConverge,string,omitempty"`
+	BandwidthPerMigration             *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
+	CompletionTimeoutPerGiB           *int64             `json:"completionTimeoutPerGiB,string,omitempty"`
+	ProgressTimeout                   *int64             `json:"progressTimeout,string,omitempty"`
+	UnsafeMigrationOverride           *bool              `json:"unsafeMigrationOverride,string,omitempty"`
+	AllowPostCopy                     *bool              `json:"allowPostCopy,string,omitempty"`
+}
+
 // setConfigFromConfigMap parses the provided config map and updates the provided config.
 // Default values in the provided config stay in tact.
 func setConfigFromConfigMap(config *v1.KubeVirtConfiguration, configMap *k8sv1.ConfigMap) error {
 	// set migration options
 	rawConfig := strings.TrimSpace(configMap.Data[MigrationsConfigKey])
 	if rawConfig != "" {
+		migrationConfig := migrationConfiguration(*config.MigrationConfiguration)
 		// only sets values if they were specified, default values stay intact
-		err := yaml.NewYAMLOrJSONDecoder(strings.NewReader(rawConfig), 1024).Decode(config.MigrationConfiguration)
+		err := yaml.NewYAMLOrJSONDecoder(strings.NewReader(rawConfig), 1024).Decode(&migrationConfig)
 		if err != nil {
 			return fmt.Errorf("failed to parse migration config: %v", err)
 		}
+		converted := v1.MigrationConfiguration(migrationConfig)
```

Comment:
I'm 99% sure this will work, but will this conversion still work if new fields are added to the v1.MigrationConfiguration that don't exist in the configMap?

Reply:
No, we still need to add at least the field to `migrationConfiguration`, we can omit the `json` if we want to ignore it. I think and hope we don't introduce new field in a time when config will be deprecated, otherwise we will need to rewrite it.